charms/jupyter-ui/src/charm.py

```diff
@@ -248,34 +248,45 @@ def _get_from_config(self, key) -> Union[OptionsWithDefault, str]:
         invalid input.
         """
         if key in IMAGE_CONFIGS:
-            return self._get_image_config(key)
+            return self._get_list_config(key)
         elif key in DEFAULT_WITH_OPTIONS_CONFIGS:
             return self._get_options_with_default_from_config(key)
         elif key == DEFAULT_PODDEFAULTS_CONFIG:
             # parsed the same as image configs
-            return self._get_image_config(key)
+            return self._get_list_config(key)
         elif key == GPU_NUMBER_CONFIG:
             return parse_gpu_num(self.model.config[key])
         else:
             return self.model.config[key]
 
-    def _get_image_config(self, key) -> OptionsWithDefault:
-        """Parse and return an image config entry, which should render to a list.
+    def _get_list_config(self, key) -> OptionsWithDefault:
+        """Parse and return a config entry which should render to a list, like the image lists.
 
-        Returns a ConfigWithDefaults with:
+        Returns a OptionsWithDefault with:
             .options: the content of the config
             .default: the first element of the list
         """
         error_message = f"Cannot parse list input from config '{key}` - ignoring this input."
         try:
             options = yaml.safe_load(self.model.config[key])
-            # Convert anything empty to an empty list
-            if not options:
+
+            # Empty yaml string, which resolves to None, should be treated as an empty list
+            if options is None:
                 options = []
+
+            # Check that we receive a list or tuple.  This filters out types that can be indexed but
+            # are not valid for this config (like strings or dicts).
+            if not isinstance(options, (tuple, list)):
+                self.logger.warning(
+                    f"{error_message}  Input must be a list or empty string. Got: {options}"
```

Comment:
So in this case, the user will understand that the config is not right when they try to use it and see that it is empty, right?

Reply:
Yeah.  So if they set `"true"` or `{}` as an input, it'll report here with this error message.  But I'm adding single quotes into the string now just to make it clear when there's an empty response

fwiw, an empty string should never actually get here (`yaml.safe_load("")` yields a `None`, not `""`), but just in case